packages/auth/src/selling-partner-api-auth.ts

```diff
@@ -27,51 +28,82 @@ export interface SellingPartnerAuthParameters {
 export class SellingPartnerApiAuth {
   public readonly accessToken: AccessToken
   public readonly sts?: SecurityTokenService
-  public readonly user?: User
+
+  private readonly _accessKeyId: string
+  private readonly _secretAccessKey: string
 
   constructor(parameters: RequireExactlyOne<SellingPartnerAuthParameters, 'refreshToken' | 'scopes'>) {
+    const clientId = parameters.clientId || process.env.LWA_CLIENT_ID
+    const clientSecret = parameters.clientSecret || process.env.LWA_CLIENT_SECRET
+    const accessKeyId = parameters.accessKeyId || process.env.AWS_ACCESS_KEY_ID
+    const secretAccessKey = parameters.secretAccessKey || process.env.AWS_SECRET_ACCESS_KEY
+    const arn = parameters.role?.arn || process.env.AWS_ROLE_ARN
+    const sessionName = parameters.role?.sessionName || process.env.AWS_ROLE_SESSION_NAME || `${pkg.name}@${pkg.version}`
+    let role = null
+
+    if (arn) {
+      role = {
+        arn,
+        sessionName
+      }
+    }
+
+    if (!clientId) {
+      throw new SellingPartnerApiAuthError('clientId is required')
+    }
+
+    if (!clientSecret) {
+      throw new SellingPartnerApiAuthError('clientSecret is required')
+    }
+
+    if (!accessKeyId) {
+      throw new SellingPartnerApiAuthError('accessKeyId is required')
+    }
+
+    if (!secretAccessKey) {
+      throw new SellingPartnerApiAuthError('secretAccessKey is required')
+    }
+
+    this._accessKeyId = accessKeyId
+    this._secretAccessKey = secretAccessKey
+
     if (parameters.refreshToken) {
       this.accessToken = new AccessToken({
-        clientId: parameters.clientId,
-        clientSecret: parameters.clientSecret,
+        clientId,
+        clientSecret,
         refreshToken: parameters.refreshToken
       })
     } else if (parameters.scopes) {
       this.accessToken = new AccessToken({
-        clientId: parameters.clientId,
-        clientSecret: parameters.clientSecret,
+        clientId,
+        clientSecret,
         scopes: parameters.scopes
       })
     } else {
       throw new TypeError('Either "refreshToken" or "scopes" must be specified')
     }
 
-    if (parameters.role) {
+    if (role) {
       this.sts = new SecurityTokenService({
-        accessKeyId: parameters.accessKeyId,
+        accessKeyId,
+        secretAccessKey,
         region: parameters.region,
-        role: parameters.role,
-        secretAccessKey: parameters.secretAccessKey
+        role
       })
     }
-
-    if (parameters.user) {
-      this.user = parameters.user
-    }
   }
 
   /**
 	 * Get AWS credentials from STS or user
 	 */
-  async getCredentials(): Promise<{
-    AccessKeyId?: string;
-    SecretAccessKey?: string;
-    SessionToken?: string;
-  } | undefined> {
+  async getCredentials(): Promise<SetOptional<Credentials, 'Expiration' | 'SessionToken'>> {
     if (this.sts) {
       return this.sts?.getCredentials()
```

Comment:
is the `?` required by ts? since there’s a check just above.

Reply:
it's not